### PR TITLE
Migrate message catch element test to use class rule

### LINKED ISSUE #2024: Message subscription close command is always send to partition 0
When closing a message subscription the command is always send to partition 0 instead of the corresponding partition id for the correlation key.

The issue is that the partition id is not set on the subscription when it is saved in the state
https://github.com/zeebe-io/zeebe/blob/aaf0be9f7d2be9787cb66e42a4429082604e6b65/broker-core/src/main/java/io/zeebe/broker/workflow/processor/CatchEventBehavior.java#L149-L156

Instead the partition is calculated before the open message subscription command is send
https://github.com/zeebe-io/zeebe/blob/aaf0be9f7d2be9787cb66e42a4429082604e6b65/broker-core/src/main/java/io/zeebe/broker/subscription/command/SubscriptionCommandSender.java#L103-L104

But on closing the message subscription we try to read the partition id from the state, which was never set
https://github.com/zeebe-io/zeebe/blob/aaf0be9f7d2be9787cb66e42a4429082604e6b65/broker-core/src/main/java/io/zeebe/broker/workflow/processor/CatchEventBehavior.java#L180

Either we calculate the partition id before setting the state, or recalculate it when closing the subscription.

### PR DESCRIPTION
**Pull request by menski**
- add await helper method to wait for records in exporter stream
- use EmbeddedBrokerRule as ClassRule in MessageCatchElementTest, reduces the runtime on my local machine from 22 seconds to 2 seconds
- replaces partition test client usage with ClientApiRule usage, as events have to be send to correct partitions, these tests only worked before as the static correlation key `order-123` hashed to partition id 0
- send close message subscription command to correct partition

related to #2014
closes #2024

### CODE REVIEW

**Review by ChrisKujawa — APPROVED** [review]
Very nice :+1: Cool when we migrate all the tests :smile: I added a new commit with deploying only once on test class start, which gain also some performance. If you like it you can squash it and merge the PR.

File: protocol-test-util/src/main/java/io/zeebe/test/broker/protocol/clientapi/ClientApiRule.java [hunk 1]
```diff
@@ -191,4 +215,98 @@ public ClientTransport getTransport() {
...
+    final int partitionId = RecordingExporter
+      .workflowInstanceRecords(WorkflowInstanceIntent.ELEMENT_ACTIVATED)
+      .withWorkflowInstanceKey(workflowInstanceKey)
+      .getFirst()
+      .getMetadata().getPartitionId();
+
+    final ExecuteCommandResponse response = createCmdRequest()
+      .partitionId(partitionId)
+      .type(ValueType.WORKFLOW_INSTANCE, WorkflowInstanceIntent.CANCEL)
+      .key(workflowInstanceKey)
```
[the remark this answers is not in the document]
Comment (menski) [reply protocol-test-util/src/main/java/io/zeebe/test/broker/protocol/clientapi/ClientApiRule.java]: Of course we should use this, totally forgot that we already have that information in the key :+1:

**Comment by menski** [comment]
bors retry

**Comment by menski** [comment]
bors retry :crossed_fingers:

**Pull request merged**